pkg/daemon/update.go

```diff
@@ -1469,12 +1469,22 @@ func (dn *Daemon) writeUnits(units []ign3types.Unit) error {
 	for _, u := range units {
 		// write the dropin to disk
 		for i := range u.Dropins {
+			dpath := filepath.Join(pathSystemd, u.Name+".d", u.Dropins[i].Name)
+
 			if u.Dropins[i].Contents == nil || *u.Dropins[i].Contents == "" {
 				glog.Infof("Dropin for %s has no content, skipping write", u.Dropins[i].Name)
-				continue
+				err := os.Remove(dpath)
```

Comment:
While I think this should work, I'm leaning towards doing it one of two ways:

1. delete this in the deleteStaleData function
2. write an empty file to the location instead of deleting it

If we do 2 I think the benefit is potentially we don't affect verification. What do you think?

Reply:
Ah, indeed. `deleteStaleData` make sense here.